test/provider_os_memory.cpp

```diff
@@ -295,3 +297,48 @@ TEST_P(umfProviderTest, purge_force_INVALID_POINTER) {
     verify_last_native_error(provider.get(),
                              UMF_OS_RESULT_ERROR_PURGE_FORCE_FAILED);
 }
+
+// Tests for allocations on numa nodes
+TEST_F(test, checkNUMAnodesAllocation) {
+    if (numa_available() == -1 || numa_num_task_nodes() <= 1) {
+        GTEST_SKIP();
+    }
+
+    umf_memory_provider_handle_t os_memory_provider = nullptr;
+    umf_os_memory_provider_params_t os_memory_provider_params =
+        UMF_OS_MEMORY_PROVIDER_PARAMS_TEST;
+
+    bitmask *nodemask = numa_parse_nodestring("1");
```

Comment:
Please move this part (creating a memory provider from a 'nodestring' to a helper function. This will be needed by other tests as well I think.

Reply:
It was replaced with `bitmask_setbit`.